lib/logstash/java_integration.rb

```diff
@@ -39,3 +39,9 @@ def self.===(other)
     super
   end
 end
+
+module java::util::Collection
+  def delete(o)
+    self.removeAll([o]) ? o : block_given? ? yield : nil
```

Comment:
because `self.remove` removes only the first item equal to `o` in `self` and this is not the Ruby   `Array#delete` contract which should deletes all items from self. `self.removeAll` does that but takes a collection of objects as argument.